bahmni-commons-omod/src/main/java/org/bahmni/module/bahmnicommons/web/v1_0/controller/search/BahmniAppointmentsPatientSearchController.java

```diff
@@ -31,14 +33,35 @@
 @RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/appointments/search/patient")
 public class BahmniAppointmentsPatientSearchController extends BaseRestController {
     
+    private PatientService patientService;
+    
+    @Autowired
+    public void setPatientService(PatientService patientService) {
+        this.patientService = patientService;
+    }
+
     @RequestMapping(method = RequestMethod.GET)
     @ResponseBody
-    public ResponseEntity<AlreadyPaged<PatientResponse>> search(HttpServletRequest request, HttpServletResponse response) throws ResponseException {
+    public ResponseEntity<AlreadyPaged<PatientResponse>> search(HttpServletRequest request,
+                                                                HttpServletResponse response) throws ResponseException {
 
         try {
             RequestContext requestContext = RestUtil.getRequestContext(request, response);
             PatientSearchParameters searchParameters = new PatientSearchParameters(requestContext);
-            List<Patient> patients = Context.getPatientService().getPatients(searchParameters.getName());
+            String query = searchParameters.getIdentifier() != null ? searchParameters.getIdentifier() : searchParameters.getName();
+            
+            if (searchParameters.getCustomAttribute() != null ||
+                    searchParameters.getAddressFieldValue() != null ||
+                    searchParameters.getPatientAttributes() != null ||
+                    searchParameters.getProgramAttributeFieldValue() != null ||
+                    searchParameters.getProgramAttributeFieldName() != null ||
+                    searchParameters.getAddressSearchResultFields() != null ||
+                    searchParameters.getPatientSearchResultFields() != null ||
+                    searchParameters.getLoginLocationUuid() != null) {
+                throw new IllegalRequestException("An unsupported search parameter was provided.");
+            }
```

Comment:
Can you not rather do
```java
if (searchParameters.getIdentifier() == null && searchParameters.getName() == null) {
  throw ...
}
```
?
And actually do that right away before trying to build `query`?

Reply:
Regarding the 2nd point you're right, it should be verified before building the query.

The snippet of code you've suggested guarantees that the search request is either asking to filter by identifier or name. But, it does not guarantee that the request is not asking for filtering for both name and ProgramAttributeFieldValue for instance. ProgramAttributeFieldValue is not supported, hence should return the exception. What do you think, does this seem right?

Reply:
@icrc-psousa you're right, we unfortunately have to check them one by one to figure out that there is an `IllegalRequestException` to throw 👍